aten/src/ATen/KinetoPluginAPI.cpp

```diff
@@ -0,0 +1,29 @@
+#include <ATen/KinetoPluginAPI.h>
+
+#include <map>
+#include <vector>
+#include <mutex>
+
+namespace at::kineto_plugin {
+
+static std::map<DeviceType, PluginFactory> plugin_factories;
+static std::mutex mutex;
+
+void SetKinetoPluginRegister(DeviceType t, PluginFactory f) {
+  std::scoped_lock<std::mutex> lock(mutex);
+  auto it = plugin_factories.find(t);
+  if (it != plugin_factories.end()) {
+    it->second = f;
+  } else {
+    plugin_factories[t] = f;
+  }
+}
```

Comment:
Should we set the priority for each register, if folloowing allocator design?
Should we provide a API to get the register?

I have no answer.

```
void SetAllocator(at::DeviceType t, at::Allocator* alloc, uint8_t priority) {
  if (priority >= allocator_priority[static_cast<int>(t)]) {
    allocator_array[static_cast<int>(t)] = alloc;
    allocator_priority[static_cast<int>(t)] = priority;
  }
}

at::Allocator* GetAllocator(const at::DeviceType& t) {
  auto* alloc = allocator_array[static_cast<int>(t)];
  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(alloc, "Allocator for ", t, " is not set.");
  return alloc;
}

```

Reply:
_Should we set the priority for each register, if folloowing allocator design?_
How can we explain the priority to the other backends?

_Should we provide a API to get the register?_
ok

Reply:
The priority to make sure the override ordering. Otherwise, override will be random, only depends on the order of initialization?